baselines/heterofl/README.md

```diff
@@ -0,0 +1,233 @@
+**Model:** This baseline uses two models:
++ Convolutional Neural Network(CNN) model is used for MNIST dataset.
++ PreResNet (preactivated ResNet) model is used for CIFAR10 dataset.
+
+These models use static batch normalization (sBN) and they incorporate a Scaler module following each convolutional layer.
+
+**Dataset:** This baseline includes MNIST and CIFAR10 datasets. 
+<table>
+  <thead>
+    <tr>
+      <th>Dataset</th>
+      <th>#classes</th>
+      <th>IID partition</th>
+      <th>non-IID partition</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>MNIST</td>
+      <td>10</td>
+      <td rowspan="2">Distribution of equal number of data examples among n clients</td>
+      <td rowspan="2">Distribution of data examples such that each client has at most 2 (customizable) classes</td>
+    </tr>
+    <tr>
+      <td>CIFAR10</td>
+      <td>10</td>
+    </tr>
+  </tbody>
+</table>
+
+**Training Hyperparameters:** 
+<table>
+  <thead>
+    <tr>
+      <th colspan="2">Description</th>
+      <th>MNIST</th>
+      <th>CIFAR10</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2">total clients</td>
+      <td colspan="2">100</td>
+    </tr>
+    <tr>
+      <td colspan="2">clients per round</td>
+      <td colspan="2">100</td>
+    </tr>
+    <tr>
+      <td colspan="2">#local epochs</td>
+      <td colspan="2">5</td>
+    </tr>
+    <tr>
+      <td rowspan="2">number of rounds</td>
+      <td>IID</td>
+      <td>200</td>
+      <td>400</td>
+    </tr>
+    <tr>
+      <td>non-IID</td>
+      <td>400</td>
+      <td>800</td>
+    </tr>
+    <tr>
+      <td colspan="2">optimizer</td>
+      <td colspan="2">SGD</td>
+    </tr>
+    <tr>
+      <td colspan="2">momentum</td>
+      <td colspan="2">5.00e-04</td>
+    </tr>
+    <tr>
+      <td colspan="2">weight-decay</td>
+      <td colspan="2">0.9</td>
+    </tr>
+    <tr>
+      <td colspan="2">learning rate</td>
+      <td>0.01</td>
+      <td>0.1</td>
+    </tr>
+        <tr>
+      <td rowspan="2">decay schedule</td>
+      <td>IID</td>
+      <td>[100]</td>
+      <td>[200]</td>
+    </tr>
+    <tr>
+      <td>non-IID</td>
+      <td>[150, 250]</td>
+      <td>[300, 500]</td>
+    </tr>
+    <tr>
+      <td colspan="2">hidden layers</td>
+      <td colspan="2">[64 , 128 , 256 , 512]</td>
+    </tr>
+  </tbody>
+</table>
+
+
+
+## Environment Setup
+
+```
+# Set python version
+pyenv install 3.10.6
+pyenv local 3.10.6
+
+# Tell poetry to use python 3.10
+poetry env use 3.10.6
+
+# install the base Poetry environment
+poetry install
+
+# activate the environment
+poetry shell
+```
+
+
+## Running the Experiments
+
+```bash  
+# The main experiment implemented in your baseline using default hyperparameters (that should be setup in the Hydra configs)
+# should run (including dataset download and necessary partitioning) by executing the command:
+
+poetry run -m heterofl.main  # Which runs the heterofl with arguments availbale in heterfl/conf/base.yaml
```

Comment:
you should either use: `poetry run python -m heterofl.main ....` or (after `poetry shell`) `python -m heterofl.main ...`. I'm inclined to present the commands as the latter (partly because it's shorter and partly because that's how most other baselines are doing it)